ci/Jenkinsfile

```diff
@@ -16,6 +16,22 @@ def abortPreviousRunUnlessMaster() {
 }
 abortPreviousRunUnlessMaster()
 
+def fileHasChanged(String filePath) {
+    def file = currentBuild.changeSets.collect({
+        it.items.collect {
+            it.affectedFiles.find { file ->
+                file.path == filePath
```

Comment:
Out of scope for this PR but: do we have any file that might get modified after any step in the CI? Maybe we can simply check for any changed file? (I say it now because maybe it is easier to implement and faster to check.)

Reply:
So like check all files? Because no file should get modified? Yea that sounds safe, but about if any files get modified, I dont know the project that well yet, we will probably see it when we run the check.

Reply:
Afaik it should never happen with our current setup (we could still define exceptions if it would be needed). It currently might happen in pre-commit with all files #261.

Reply:
Should we maybe add pre-commit next to CI? Then it won't even make it to that point and will fail before.

Reply:
What do you mean with "next to CI"?

Reply:
There is currently no `pre-commit run --all-files` check inside our pipeline

Reply:
Sorry, I phrased it a bit poorly. I meant "Should we add, as next step, pre-commit to our pipeline"

Reply:
> Sorry, I phrased it a bit poorly. I meant "Should we add, as next step, pre-commit to our pipeline"

I think we first should get all the open PRs merged :wink: 

Then we can create some meta-issue and do prioritization there. Didn't you already create such a issue? Prioritization discussions should happen there.

> The idea is to have a static analysis stage, which will test all sorts of language formatting and run a pre-commit at the end.

Great. The stages do not really need to make sense. Their main goal is to provide good parallelization (utilization of our infrastructure).